scripts/release.py

```diff
@@ -206,16 +206,237 @@ def create_draft_release(
+                if author:
+                    author = author.split(" ")
+                    author_slack = "@" + author[0] + "." + author[-1]
+                    author_slack_handles.append(author_slack)
+                    author_dd = author_slack + "@datadoghq.com"
+                pr_num = re.findall(r"\(#(\d{4})\)", commit.commit.message)[0]
+                url = "https://github.com/DataDog/dd-trace-py/pull/{pr_num}".format(pr_num=pr_num)
+                prs_details.append({"author_dd": author_dd, "url": url, "rn_piece": rn_piece})
+
+    # edit release notes to be put inside notebook
+    rn = rn.replace("\n-", "\n- [ ]")
+    for pr_details in prs_details:
+        rn_piece = pr_details["rn_piece"]
+        i = rn.rfind(rn_piece)
+        if i != -1:
+            e = i + len(rn_piece)
+            # check to make sure there was a match
+            rn = (
+                rn[:e]
+                + "\nPR:{pr}\nTester: {author_dd}".format(pr=pr_details["url"], author_dd=pr_details["author_dd"])
+                + rn[e:]
+            )
+
+    # create the review notebook and add the release notes formatted for testing
+    # get notebook template
+    template_notebook = subprocess.check_output(
+        'curl -X GET "https://api.datadoghq.com/api/v1/notebooks/4888117" \
+    -H "Accept: application/json" \
+    -H "DD-API-KEY: {dd_api_key}" \
+    -H "DD-APPLICATION-KEY: {dd_app_key}"'.format(
+            dd_api_key=dd_api_key, dd_app_key=dd_app_key
+        ),
+        shell=True,
+        cwd=os.pardir,
+    )
+    data = template_notebook.decode("utf8")
+    data = json.loads(data)
+    # change the text inside of our template to include release notes
+    data["data"]["attributes"]["cells"][1]["attributes"]["definition"]["text"] = (
+        "#  Release notes to test\n%s\n<Tester> \n<PR>\n\n\n## Release Notes that will not be tested\n- <any release notes for PRs that don't need manual testing>\n\n\n"  # noqa
+        % (rn)
+    )
+    # grab the latest commit id on 1.x to mark the rc notebook with
+    main_branch = dd_repo.get_branch(branch="1.x")
+    commit_id = main_branch.commit
+
+    # pull the cells out to be transferred into a new notebook
+    cells = data["data"]["attributes"]["cells"]
+    nb_name = "ddtrace-py testing %s commit %s" % (name, commit_id.sha)
+    interpolated_notebook = {
+        "data": {
+            "attributes": {
+                "cells": cells,
+                "name": nb_name,
+                "time": {"live_span": "1d"},
+            },
+            "type": "notebooks",
+        }
+    }
+
+    notebook_json = json.dumps(interpolated_notebook, indent=4, sort_keys=True)
+
+    headers = {
+        "Accept": "application/json",
+        "Content-Type": "application/json",
+        "DD-API-KEY": dd_api_key,
+        "DD-APPLICATION-KEY": dd_app_key,
+    }
+    # create new release notebook
+    requests.post("https://api.datadoghq.com/api/v1/notebooks", data=notebook_json, headers=headers)
```

Comment:
can we not create a notebook with the API client?

Reply:
The api client doesn't provide a nice way to pull a notebook, modify a certain aspect of it, and then create a new notebook with the modified notebook. That's what I tried at first and it was quite difficult.